### PR TITLE
Add trigger bit to emulated GlobalExtBlk to indicate event was not selected …

### PR DESCRIPTION
**Pull request by apana**
…due to prefiring of the trigger

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

File: L1Trigger/L1TGlobal/plugins/L1TExtCondProducer.cc [hunk 1]
```diff
@@ -82,8 +90,26 @@ using namespace l1t;
...
     setBptxPlus_ (iConfig.getParameter<bool>("setBptxPlus")),
     setBptxMinus_ (iConfig.getParameter<bool>("setBptxMinus")),
-    setBptxOR_ (iConfig.getParameter<bool>("setBptxOR"))
+    setBptxOR_ (iConfig.getParameter<bool>("setBptxOR")),
+    tcdsInputTag_(iConfig.getParameter<edm::InputTag>("tcdsRecordLabel"))
   {
+
+    makeTriggerRulePrefireVetoBit_=false;
+
+    m_triggerRulePrefireVetoBit=255;
```
Comment (fabiocos) [inline L1Trigger/L1TGlobal/plugins/L1TExtCondProducer.cc]: @apana @rekovic this magic value looks related to https://github.com/cms-sw/cmssw/blob/master/DataFormats/L1TGlobal/interface/GlobalExtBlk.h#L47 and now everything looks consistent. But why having the two as independent, and using a LogWarning that risks to go unnoticed? If this has to be the last bit of the array just use that position no? Or I would escalate the severity to LogError at least (although still it does not seem the best solution to me...)

**Comment by peruzzim** [comment]
Looking at the GlobalExtBlk constructor here:
https://github.com/cms-sw/cmssw/blob/e5aead7a3343499ddf14e980eceb36fe6a767ffe/DataFormats/L1TGlobal/src/GlobalExtBlk.cc#L37
one would need to have at least a method to get the size of the `m_extDecision` vector:
https://github.com/cms-sw/cmssw/blob/e5aead7a3343499ddf14e980eceb36fe6a767ffe/DataFormats/L1TGlobal/interface/GlobalExtBlk.h#L65
otherwise it's impossible to reverse-engineer the logic you use to set the 255-or-max-minus-1 bit of that vector.

**Comment by fabiocos** [comment]
+operations

the previous circular dependency of tcdsDigis and rawDataCollector looks to have been solved,, and the TauEmbedding previously failing unit test now works smoothly

**Comment by fabiocos** [comment]
+1

**Comment by rekovic** [comment]
@peruzzim 
These comments are valid: avoiding setting the size by hand here and having a  public accessor which is transparent to the position of this bit.  

@apana 
Not to introduce delay in integration of the PR now, after this PR is merged (soon), we need to follow up with another PR which provide the above functionality.

**Comment by apana** [comment]
Dear @peruzzim and @rekovic,

I agree this could have been coded better. I will try to put in the follow up PR asap.

**Pull request merged**